maat-court-data-api/src/main/java/gov/uk/courtdata/assessment/impl/FinancialAssessmentImpl.java

```diff
@@ -1,56 +1,83 @@
 package gov.uk.courtdata.assessment.impl;
 
+import gov.uk.courtdata.assessment.mapper.FinancialAssessmentMapper;
+import gov.uk.courtdata.dto.FinancialAssessmentDTO;
 import gov.uk.courtdata.entity.FinancialAssessmentEntity;
 import gov.uk.courtdata.enums.FinancialAssessmentType;
-import gov.uk.courtdata.exception.MAATCourtDataException;
 import gov.uk.courtdata.repository.FinancialAssessmentRepository;
 import gov.uk.courtdata.repository.HardshipReviewRepository;
 import gov.uk.courtdata.repository.PassportAssessmentRepository;
 import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.stereotype.Service;
-import org.springframework.transaction.annotation.Transactional;
 
 @Service
 @RequiredArgsConstructor
 @Slf4j
 public class FinancialAssessmentImpl {
 
+    private final FinancialAssessmentMapper assessmentMapper;
     private final PassportAssessmentRepository passportAssessmentRepository;
     private final HardshipReviewRepository hardshipReviewRepository;
     private final FinancialAssessmentRepository financialAssessmentRepository;
 
-    public FinancialAssessmentEntity getAssessment(Integer financialAssessmentId) {
+    public FinancialAssessmentEntity find(Integer financialAssessmentId) {
         return financialAssessmentRepository.getById(financialAssessmentId);
     }
 
-    public FinancialAssessmentEntity updateAssessment(FinancialAssessmentEntity financialAssessment) {
-        if (financialAssessment.getFullAssessmentDate() != null) {
-            financialAssessment.setAssessmentType(FinancialAssessmentType.FULL.getValue());
+    public FinancialAssessmentEntity update(FinancialAssessmentDTO financialAssessment) {
+        FinancialAssessmentEntity existingAssessment = financialAssessmentRepository.getById(financialAssessment.getId());
+        existingAssessment.setFassInitStatus(financialAssessment.getFassInitStatus());
```

Comment:
any specific reason for now using builder here?

Reply:
Unfortunately, using a mapper here will not work - the null values in the new object don't get updated after saving due to limitations in the EntityManager merge method which gets called when executing a save operation when there is an existing record.